.github/workflows/release.yaml

```diff
@@ -334,3 +341,60 @@ jobs:
             channel_id: ${{ inputs.slack-channel-id }}
             title: '[${{ steps.component-descriptor.outputs.component-version }} released]'
             file: /tmp/release-notes/slack-release-notes.md
+
+      - name: reset-repository-to-release-commit
+        if: ${{ inputs.create-release-branch == 'true' && inputs.next-version != 'bump-patch' }}
+        shell: bash
+        run: |
+          set -euo pipefail
+
+          # version may be substituted in github-tag-template
+          version="${{ steps.component-descriptor.outputs.version }}"
+          tag_name="${{ inputs.github-tag-template }}"
+
+          git fetch origin
+          git checkout "$tag_name"
+
+      - name: create-release-branch-bump-commit
+        if: ${{ inputs.create-release-branch == 'true' && inputs.next-version != 'bump-patch' }}
+        uses: gardener/cc-utils/.github/actions/version@master
+        with:
+          versionfile: ${{ inputs.version-versionfile }}
+          read-callback: ${{ inputs.version-read-callback }}
+          write-callback: ${{ inputs.version-write-callback }}
+          callback-action-path: ${{ inputs.next-version-callback-action-path }}
+          commit-message: "next version: ${version}"
+          version-operation: bump-patch
+          prerelease: dev
+          repository-operation: commit-to-head
+          commit-kind: bump
+
+      - name: create-and-push-release-branch
+        if: ${{ inputs.create-release-branch == 'true' && inputs.next-version != 'bump-patch' }}
+        shell: bash
+        run: |
+          set -euo pipefail
+
+          version="${{ steps.component-descriptor.outputs.version }}"
+
+          version_no_v="${version#v}"
+
+          export major=$(echo "$version_no_v" | cut -d. -f1)
+          export minor=$(echo "$version_no_v" | cut -d. -f2)
+          export patch=$(echo "$version_no_v" | cut -d. -f3)
+
+          # default values if not present in .ocm/branch-info.yaml file
+          release_branch_template="release-v$major.$minor"
+
+          if [ -f .ocm/branch-info.yaml ]; then
+            release_branch_template_overwrite=$(cat .ocm/branch-info.yaml | yq .release-branch-template)
+
+            if [ $release_branch_template_overwrite != "null" ]; then
+              echo "INFO: Found .ocm/branch-info.yaml file with branch template $release_branch_template_overwrite"
+              release_branch_template=$release_branch_template_overwrite
+            fi
+          fi
+
+          branch="$(echo $release_branch_template | envsubst)" # fill in values of major/minor/patch
```

Comment:
nice! did not know about envsubst. I suppose we should consider switching other places in code (e.g. in `version`-action)